### DISCUSSION TITLE
How to change font style in VS Code

### DISCUSSION THREAD

**Discussion by rahulmasal** [Mark2Mark/gintronic#15] [Q&A] [answered]
in the Gintronic font family,, I need to try many other styles, but I don't know how to apply those styles in vs code settings kindly guide me if possible!

**Comment by Mark2Mark** [comment]
Hi @rahulmasal,

Sure. Did you have a look at [this discussion](https://github.com/Mark2Mark/gintronic/discussions/11)? The first entry shows already how to do it in VSCode.

That topic focusses on OpenType features, however, there is also the basic font setting shown.

In the VSCode user settings (`settings.json`), you need to set the
```
"editor.fontFamily": "Gintronic"
```

This will usually pick the Regular style, and also its Italic pendant (if installed and used by a theme).

You can also specigfy the desired font style:
```
"editor.fontFamily": "Gintronic-Regular"
```
(must be with hyphen, depends on the font as it seems.)

Save the settings.json file. The changes should take effect immediately.

If you want to define fallbacks, you name the fonts in a chain, where the next one is used in case the first is not installed, and so on:

```
"editor.fontFamily": "Gintronic, Consolas, monospace",
```

**Please note** that VS Code does not support font styles like bold or italic for the entire editor. You can only change the font family. However, syntax highlighting will still use different font styles for different parts of your code.


Let me know if you still need assistance.

**Reply by Mark2Mark** [reply]
What do you mean "vs code not support styles"? It does indeed.

I double checked, and it needs to be with hyphen (edited my instruction above, though it may depend on the font).

Are you on windows?
This is how it behaves on a Mac:


https://github.com/Mark2Mark/gintronic/assets/5824595/a6d33fc2-63b4-4ce7-bb03-e6ab96426112

(in this video, one can see some extra quotes ("'Gintronic', ..."), those are not necessary. I just had them sticking around from trying if without hyphen also works)

**Reply by Mark2Mark** [reply]
In the video where you put it in the settings, it does not even apply the font, when you just have "Gintronic" (no style, just the family name). That is odd, as it should use the font in that case.

1. I assume you saved after the edits of the settings file?
1. Is it the user settings.json?
1. Can we double check if the fonts are properly loaded by the system? Here are some ways to check this:

#### A) Control Panel:

Open the Control Panel on your Windows computer.
Navigate to "Appearance and Personalization" and then "Fonts."
The Fonts window will display a list of all installed fonts.

#### B) File Explorer:

Open File Explorer and navigate to C:\Windows\Fonts (or %windir%\Fonts).
This folder contains all installed fonts. You can visually inspect the list to see if the specific font is present.

#### C) Settings App:

Open the Settings app in Windows.
Go to "Personalization" and then "Fonts." This section will show you a preview of each installed font.

#### D) Word Processor or Text Editor:

Open a word processor or text editor that allows you to change the font.
Look for the font in the font selection menu. If it's listed, then it's installed.

#### E) Task Manager:

Open the Task Manager (you can right-click on the taskbar and select "Task Manager").
Go to the "Processes" tab.
Look for a process related to font management. This can sometimes give you an idea of the fonts installed.

**Comment by rahulmasal** [comment]


**Reply by Mark2Mark** [reply]
Thank you.

From your video we can summarize that so far these work:
```
Gintronic
Gintronic-Regular
Gintronic-Medium
Gintronic-Black
```
Also the Italic is already applied, as seen in these working weights, where for example your dictionary keys are set in italic and the values in upright. That is a syntax highlighting (theme) setting, where certain words are italisized.

Can you please answer my questions that I asked you? Especially question 3: can you make sure all styles are properly installed? Apparently, they need to be installed for all users on Windows in order for VSCode to see them.

**Reply by Mark2Mark** [reply]
Thank you. Do you have the .ttf font files or .otf?

Can you send me an email with your invoice? Then I can send you the .ttf files, maybe they could work on Windows.

**Reply by Mark2Mark** [reply]
- Does it work woth other font families?
- Did I already send you the ttf files? What is your order number, please?

**Reply by Mark2Mark** [reply]
Can you please tell me your order number? Thank you.

**Comment by Mark2Mark** [comment, marked as answer]
@rahulmasal, have you removed your comments and questions because I requested your order number? To assist you better with the issue you're experiencing, I need that number. It seems to be a unique problem, not reported by other users.

I'll assume you're using a pirated version of Gintronic until you provide the order number. Using pirated fonts is not only unethical but also beyond my control for both support and responsibility for the quality of those fonts. Your cooperation in providing the order number would be appreciated.

**Reply by rahulmasal** [reply]
Sir,

i never use pirated softwares or material font given by my company team member to use for code, but i now switched to jetbrains mono which is supporting properly in vs code.

**Reply by Mark2Mark** [reply]
So that is why you deleted all your posts here?

> i never use pirated softwares or material

<img width="931" alt="Screenshot 2023-12-01 at 21 15 03" src="https://github.com/Mark2Mark/gintronic/assets/5824595/99c83e93-556d-4937-8556-b0b63f855bcf">

I think it is not fair to the designers/developers who make the tools you use to share these tools without authorization withing the company.

**Reply by Mark2Mark** [reply]
> ... font given by my company team member to use for code, but i now switched to jetbrains mono which is supporting properly in vs code.

Well, the non-pirated version of Gintronic works just fine in VSCode. You probably used a ripped font, which usually breaks a lot of functionality of the fonts.

**Reply by rahulmasal** [reply]
Sir,

Thanks for checked my recent activity, You are Awesome, but i switched to jetbrains mono fonts.
